### ISSUE TITLE
Unable to set custom root token in dev mode

### ISSUE THREAD

**Issue by hpio** [hashicorp/vault-helm#414]
**Describe the bug**
When deploying vault in dev mode I am unable to set a custome VAULT_DEV_ROOT_TOKEN_ID as the variable is duplicated in the resulting statefulset

**To Reproduce**
1. Install chart and pass the following values:
```
  replicaCount: 1
  server:
    dev:
      enabled: true
    standalone:
      listener:
        tcp:
          address: "[::]:8200"
          cluster_address: "[::]:8201"
          tls_disable: 1
        ui: true
      enabled: true
    extraSecretEnvironmentVars:
      - envName: VAULT_DEV_ROOT_TOKEN_ID
        secretName: vault
        secretKey: appToken
```
2. look at yaml file of vault statefulset
```
        - name: VAULT_DEV_ROOT_TOKEN_ID
          value: root
        - name: VAULT_DEV_ROOT_TOKEN_ID
          valueFrom:
            secretKeyRef:
              key: appToken
              name: vault
```

VAULT_DEV_ROOT_TOKEN_ID has been added twice. This is because `_helpers.tpl` has the following function:
```
{{- define "vault.envs" -}}
  {{ if eq .mode "dev" }}
            - name: VAULT_DEV_ROOT_TOKEN_ID
              value: "root"
  {{ end }}
{{- end -}}
```

Other useful info to include: vault pod logs, `kubectl describe statefulset vault` and `kubectl get statefulset vault -o yaml` output
`kubectl describe statefulset vault`

```
❯ kubectl describe statefulset uat-i234-vault
Name:               uat-i234-vault
Namespace:          uat-i234
CreationTimestamp:  Wed, 18 Nov 2020 10:09:09 +0000
Selector:           app.kubernetes.io/instance=uat-i234,app.kubernetes.io/name=vault,component=server
Labels:             app.kubernetes.io/instance=uat-i234
                    app.kubernetes.io/managed-by=Helm
                    app.kubernetes.io/name=vault
Annotations:        meta.helm.sh/release-name: uat-i234
                    meta.helm.sh/release-namespace: uat-i234
Replicas:           1 desired | 1 total
Update Strategy:    OnDelete
Pods Status:        1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app.kubernetes.io/instance=uat-i234
                    app.kubernetes.io/name=vault
                    component=server
                    helm.sh/chart=vault-0.8.0
  Service Account:  uat-i234-vault
  Containers:
   vault:
    Image:       vault:1.5.4
    Ports:       8200/TCP, 8201/TCP, 8202/TCP
    Host Ports:  0/TCP, 0/TCP, 0/TCP
    Readiness:   exec [/bin/sh -ec vault status -tls-skip-verify] delay=5s timeout=3s period=5s #success=1 #failure=2
    Environment:
      HOST_IP:                   (v1:status.hostIP)
      POD_IP:                    (v1:status.podIP)
      VAULT_K8S_POD_NAME:        (v1:metadata.name)
      VAULT_K8S_NAMESPACE:       (v1:metadata.namespace)
      VAULT_ADDR:               http://127.0.0.1:8200
      VAULT_API_ADDR:           http://$(POD_IP):8200
      SKIP_CHOWN:               true
      SKIP_SETCAP:              true
      HOSTNAME:                  (v1:metadata.name)
      VAULT_CLUSTER_ADDR:       https://$(HOSTNAME).uat-i234-vault-internal:8201
      HOME:                     /home/vault
      VAULT_DEV_ROOT_TOKEN_ID:  root
      VAULT_DEV_ROOT_TOKEN_ID:  <set to the key 'appToken' in secret 'vault'>  Optional: false
    Mounts:
      /home/vault from home (rw)
  Volumes:
   home:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
Volume Claims:  <none>
Events:         <none>
```
**Expected behavior**
VAULT_DEV_ROOT_TOKEN_ID should not be set by default in dev mode unless specified

**Environment**
* Kubernetes version: 
  * EKS
* vault-helm version: 0.8.0

**Comment by hpio** [comment]
created a PR that fixes the issue https://github.com/hashicorp/vault-helm/pull/415

**Comment by jasonodonnell** [comment]
Fixed via #415
